src/test/java/com/checkmarx/flow/cucumber/integration/end2end/genericendtoend/Repository.java

```diff
@@ -669,39 +661,186 @@ void verifyPRUpdated(String engine) {
+        {
+            String projectName = repo;
+            String getProjectsUrl = String.format("%s%s%s", gitLabProperties.getApiUrl(), GET_PROJECT_URL, projectName);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(getProjectsUrl, HttpMethod.GET, httpEntity, String.class);
+
+            JSONArray jsonArray = new JSONArray(response.getBody());
+            JSONObject obj = jsonArray.getJSONObject(0);
+
+            int responseProjectId = obj.getInt("id");
+            log.info("found project Id: '{}' for project '{}'", responseProjectId, projectName);
+            return  obj.getInt("id");
+        }
+
+        @Override
+        void generateHook(HookType hookType) {
+
+            log.info("creating new webhook to project '{}'", projectId);
+            String hooksUrl = String.format("%s%s", gitLabProperties.getApiUrl(), CREATE_WEBHOOK_URL);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(hooksUrl, HttpMethod.POST, httpEntity, String.class, projectId, hookTargetURL, gitLabProperties.getWebhookToken());
+            JSONObject obj = new JSONObject(response.getBody());
+            webhookId = obj.getInt("id");
+            log.info("webhook created with Id: '{}'", webhookId);
+        }
+
+        private void pushFileToBranch(String textContent, String path){
+            log.info("pushing file '{}' tp project '{}' ", path, projectId);
+
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("branch", "master");
+            jsonObject.put("commit_message", "pushing file");
+            JSONArray jsonArray = new JSONArray();
+            JSONObject actionObject = new JSONObject();
+            actionObject.put("action", "create");
+            actionObject.put("file_path", path);
+            actionObject.put("content", textContent);
+            jsonArray.put(actionObject);
+            jsonObject.put("actions", jsonArray);
+
+
+            String commitUrl = String.format("%s%s", gitLabProperties.getApiUrl(), COMMIT_URL);
+            HttpEntity<String> httpEntity = new HttpEntity<>(jsonObject.toString(), getHeaders());
+
+            ResponseEntity<String> response = restTemplate.exchange(commitUrl, HttpMethod.POST, httpEntity, String.class, projectId);
+            JSONObject obj = new JSONObject(response.getBody());
+        }
+
+        @Override
+        void pushFile(String base64content, String textContent) {
+            pushFileToBranch(textContent, CODE_FILE_PATH);
+        }
+
+        @Override
+        void generateConfigAsCode(GenericEndToEndSteps genericEndToEndSteps) {
+            try{
+                String content = genericEndToEndSteps.getConfigAsCodeTextContent(GITLAB_CONFIG_AS_CODE_FILE);
+                pushFileToBranch(content, GITLAB_CONFIG_AS_CODE_PATH);
```

Comment:
For a method called pushFileToBranch, I would expect a branch name as one of the arguments.